### ISSUE TITLE
Support stage-3 class static blocks

### ISSUE THREAD

**Issue by sosukesuzuki** [swc-project/swc#2125]
**Describe the feature**

Support Stage-3 proposal class static blocks.

**Babel plugin or link to the feature description**

https://babeljs.io/docs/en/babel-plugin-proposal-class-static-block

**Additional context**

TypeScript 4.4 supports class static blocks (https://devblogs.microsoft.com/typescript/announcing-typescript-4-4-rc/#static-blocks)

**Comment by sosukesuzuki** [comment]
@kdy1 If this feature is needed, can I work on this?

**Comment by kdy1** [comment]
Of course. Thank you a lot!

**Comment by dsherret** [comment]
@sosukesuzuki have you started working on this? I'm eager to support it given that TS 4.4 is now out. If not, I can start working on it on Monday.

**Comment by sosukesuzuki** [comment]
I've started to work! I'll do the rest of the work today.

**Comment by dsherret** [comment]
@sosukesuzuki great work. Thanks for implementing this, @sosukesuzuki! 🙂

**Comment by alexander-akait** [comment]
@kdy1 What we should finish here?

**Comment by kdy1** [comment]
We should implement transforms for static blocks so the code can be used while targeting old browsers.

**Comment by sosukesuzuki** [comment]
@kdy1 Is there any documentation useful for writing transforms? I've read the code for a while, but it's difficult...

**Comment by alexander-akait** [comment]
Yep, I am faced with this too, because code base is big (some of use is new in rust), maybe will be great to create small examples for popular cases, like replace `Hello world!` on `Hello swc!`.

**Comment by kdy1** [comment]
@sosukesuzuki @alexander-akait 
Maybe documentation at https://docs.rs/swc_visit/0.2.6/swc_visit/ would be useful.
If it's not enough, please tell me. I want to improve documentations, but it's hard to do it by myself.
